src/pybind/mgr/cephadm/serve.py

```diff
@@ -1152,97 +1144,90 @@ def _run_cephadm(self,
 
         :env_vars: in format -> [KEY=VALUE, ..]
         """
+
+        CephadmServe.loop.run_until_complete(self.ssh._remote_connection(host, addr))
```

Comment:
`run_until_complete` should only be called within the `ssh.py`. I.e. `ssh.py` should expose an interface that is easy to use and doesn't require the users of this module to care about this